test/asynchronous/test_client_bulk_write.py

```diff
@@ -577,3 +583,22 @@ async def test_timeout_in_multi_batch_bulk_write(self):
             if event.command_name == "bulkWrite":
                 bulk_write_events.append(event)
         self.assertEqual(len(bulk_write_events), 2)
+
+
+class TestClientBulkWriteMock(AsyncMockClientTest):
+    @async_client_context.require_version_min(8, 0, 0, -24)
+    async def test_handles_non_pymongo_error(self):
+        mock_client = await AsyncMockClient.get_async_mock_client(
```

Comment:
Instead of using a mock client, what do you think about using a real client with unittest.mock.patch on _AsyncClientBulk.write_command (or even Connection.write_command)?

Reply:
Sure I can change that!